setup.py

```diff
@@ -794,6 +800,16 @@ def run(self):
     elif sys.platform == 'darwin':
         ext_modules.append(Extension('borg.platform.darwin', [platform_darwin_source]))
 
+    try:
+        ncpus = __import__("multiprocessing").cpu_count()
+    except ImportError:
+        ncpus = 1
+
+    if cython_available:
+        from Cython.Build import cythonize
+        # compile .pyx extensions to .c in parallel
+        ext_modules = cythonize(ext_modules, nthreads=(ncpus+1))
```

Comment:
is it safe to use nthreads > 1 or can there be race conditions or other causes that make cythonize fail?

formatting: `nthreads=ncpus + 1`

Reply:
I can't say it's safe for sure, but my understanding is that it's pretty much the same as 'make -j' as cython is compiled similarly to a normal C project, i.e., it's embarrassingly parallel/no shared state and so on. The implementation uses multiprocessing.Pool.map_async to compile in multiple processes.